pkg/common/configs/configvalidator.go

```diff
@@ -357,13 +357,32 @@ func checkLimit(limit Limit, currIdx int, userWildCardIdx, groupWildCardIdx *int
 	}
 	// at least some resource should be not null
 	if limit.MaxApplications == 0 && resources.IsZero(limitResource) {
-		return fmt.Errorf("invalid resource combination for limit names '%s' all resource limits are null", limit.Users)
+		return fmt.Errorf("invalid resource combination for limit user names '%s' groups %s all resource limits are null", limit.Users, limit.Groups)
 	}
+
+	if queue.MaxApplications != 0 && (limit.MaxApplications > queue.MaxApplications || limit.MaxApplications == 0) {
+		return fmt.Errorf("invalid MaxApplications settings for limit user names '%s' groups %s exeecd current the queue MaxApplications", limit.Users, limit.Groups)
+	}
+
+	// If queue is RootQueue, the queue.Resources.Max will be null, we don't need to check for root queue
+	// But we may need to check the root resource during loading the config and after partition resource loading when update node
+	if queue.Name != RootQueue {
+		queueMaxResource, err := resources.NewResourceFromConf(queue.Resources.Max)
+		if err != nil {
+			log.Logger().Debug("resource parsing failed",
+				zap.Error(err))
+			return fmt.Errorf("parse queue max resource failed: %s", err.Error())
```

Comment:
Can we have queue name in this error message?

Reply:
Sure @manirajv06 , thank you for review, addressed in latest PR.